c7n/reports/csvout.py

```diff
@@ -124,15 +124,18 @@ def uniq_by_id(self, records):
         return uniq
 
     def to_csv(self, records, reverse=True):
-        filtered = filter(self.filter_record, records)
-        log.debug("Filtered from %d to %d" % (len(records), len(filtered)))
-        if 'CustodianDate' in records[0]:
-            filtered.sort(
-                key=lambda r: r['CustodianDate'], reverse=reverse)
-        uniq = self.uniq_by_id(filtered)
-        log.debug("Uniqued from %d to %d" % (len(filtered), len(uniq)))
-        rows = map(self.extract_csv, uniq)
-        return rows
+        if len(records) > 0:
```

Comment:
rather than indenting the whole block, short circuit the other case, ie. if not records: return []  and then do the toher case.